pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/loader/invertedindex/FSTIndexHandler.java

```diff
@@ -119,7 +125,9 @@ private void createFSTIndexForColumn(ColumnMetadata columnMetadata)
     String segmentName = _segmentMetadata.getName();
     String column = columnMetadata.getColumnName();
     File inProgress = new File(_indexDir, column + ".fst.inprogress");
-    File fstIndexFile = new File(_indexDir, column + FST_INDEX_FILE_EXTENSION);
+    String fileExtension = _fstType == FSTType.LUCENE ? FST_INDEX_FILE_EXTENSION
```

Comment:
We should use the same file extension as we are treating them as the same index but different versions